src/test/java/org/openrewrite/java/migrate/lombok/LombokValueToRecordTest.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lombok;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+import org.openrewrite.test.TypeValidation;
+
+import static org.openrewrite.java.Assertions.java;
+import static org.openrewrite.java.Assertions.version;
+
+public class LombokValueToRecordTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new LombokValueToRecord(false))
+          .parser(JavaParser.fromJavaVersion().classpath("lombok"));
+    }
+
+    @Test
+    void valueAnnotatedClassWithUseExactOptionKeepsLombokToString() {
+        //language=java
+        rewriteRun(
+          s -> s.recipe(new LombokValueToRecord(true)),
+          version(
+            java(
+              """
+                import lombok.Value;
+                                
+                @Value
+                public class Test {
+                    String field1;
+                    
+                    String field2;
+                }
+                """,
+              """
+                public record Test(
+                    String field1,
+                    
+                    String field2) {
+                    @Override
+                    public String toString() {
+                        return "Test(" +
+                                "field1=" + field1 + ", " +
+                                "field2=" + field2 +
+                                ")";
+                    }
+                }
+                """),
+            17
+          )
+        );
+    }
+
+    @Test
+    void convertOnlyValueAnnotatedClassWithoutDefaultValuesToRecord() {
+        //language=java
+        rewriteRun(
+          // TODO: find a way to please type validation so this workaround is not required anymore
+          s -> s.typeValidationOptions(TypeValidation.none()),
```

Comment:
Probably best to get this resolved before we merge, such that subsequent recipes don't trip over missing types; let us know if you'd need help to resolve the missing types!

Reply:
I would be really happy to get support regarding this. At the moment i suspect in the test somehow even the types are in the same package somehow the environment does not know the adjusted types after the first run.

Reply:
It seems you've worked through most of the review feedback, such that there's less chance for conflicts if we were to step in to help out here. Do you still need help? Typically I'd step through with the debugger and try things that way, but not sure yet as I haven't looked

Reply:
So the type information is already present on the input, and we're reusing the method invocations from before. As such I don't think we need to hold up this PR over missing type information, as a next run of OpenRewrite using the same sources would actually work. We can leave the missing type information as a nice to have for a future enhancement, with limited use.